client-tools/Shell/TelemetricsHandler.cpp

```diff
@@ -0,0 +1,348 @@
+    ClientManager clientManager(
+        _server.getFeature<HttpEndpointProvider, ClientFeature>(),
+        Logger::FIXME);
+    std::unique_lock lk(_mtx);
+
+    _httpClient = clientManager.getConnectedClient(true, false, false, 0);
+    if (_httpClient != nullptr && _httpClient->isConnected()) {
+      auto& clientParams = _httpClient->params();
+      clientParams.setRequestTimeout(30);
+      lk.unlock();
+      std::unique_ptr<httpclient::SimpleHttpResult> response(
+          _httpClient->request(rest::RequestType::GET, url, nullptr, 0));
+      lk.lock();
+
+      result = this->checkHttpResponse(response);
+      if (auto errorNum = result.errorNumber();
+          errorNum == TRI_ERROR_NO_ERROR ||
+          errorNum == TRI_ERROR_HTTP_FORBIDDEN) {
+        _telemetricsFetchedInfo.add(response->getBodyVelocyPack()->slice());
+        _telemetricsFetchResponse = std::move(result);
+        break;
+      } else {
+        _telemetricsFetchResponse = std::move(result);
+      }
+    }
+    _runCondition.wait_for(lk, std::chrono::seconds(timeoutInSecs),
+                           [this] { return _server.isStopping(); });
+    timeoutInSecs *= 2;
+    timeoutInSecs = std::min<uint32_t>(100, timeoutInSecs);
+  }
+}
+
+// Sends telemetrics to the endpoint and ests the redirection when telemetrics
+// is sent to an endpoint, though not exactly the same thing, because here the
+// endpoint is local, hence the same client is used and a new endpoint is not
+// created.
+std::optional<VPackBuilder> TelemetricsHandler::sendTelemetricsToEndpoint(
+    std::string const& reqUrl) {
+  VPackBuilder testResponseBuilder;
+
+  bool isLocal = false;
+  if (reqUrl.starts_with('/')) {
+    isLocal = true;
+  }
+  bool isOriginalUrl = (reqUrl.compare(kOriginalUrl) == 0) ? true : false;
+
+  std::string url = reqUrl;
+
+  Result result = {TRI_ERROR_NO_ERROR};
+  uint32_t timeoutInSecs = 10;
+  uint64_t sslProtocol = TLS_V13;
+  uint32_t maxRedirects = 5;
+
+  std::string const body = _telemetricsFetchedInfo.toJson();
+
+  uint32_t numRedirects = 0;
+  basics::StringBuffer compressedBody;
+
+  while (!_server.isStopping()) {
+    std::unique_lock lk(_mtx, std::defer_lock);
+    std::unordered_map<std::string, std::string> headers;
+    headers.emplace(StaticStrings::ContentTypeHeader,
+                    StaticStrings::MimeTypeJson);
+    std::string relativeUrl;
+    if (!isOriginalUrl) {
+      headers.emplace(StaticStrings::ContentLength,
+                      std::to_string(compressedBody.size()));
+    } else {
+      auto res =
+          encoding::gzipCompress(reinterpret_cast<uint8_t const*>(body.data()),
+                                 body.size(), compressedBody);
+      if (res != TRI_ERROR_NO_ERROR) {
+        break;
+      }
+      headers.emplace(StaticStrings::ContentLength,
+                      std::to_string(body.size()));
+      headers.emplace(StaticStrings::ContentEncoding,
+                      StaticStrings::EncodingGzip);
+    }
```

Comment:
Why is gzip compression only used for the originalUrl and not for other URLs?
If there is a rationale behind this, there should be a code comment explaining the reason.